File: 20190209-java-phone-numbers/src/main/java/com/katas/PhoneList.java
package com.katas;

import java.util.List;

class PhoneList {
    private List<String> phones;

    PhoneList(List<String> phones) {
        this.phones = phones;
    }

    boolean isConsistent() {
        for (var i = 0; i < phones.size(); i++) {
            for (var z = 0; z < phones.size(); z++) {
                if (i == z) continue;

                if (phones.get(i).contains(phones.get(z))) {
                    return false;
                }
            }
        }

        return true;
    }
}

File: 20181125-kotlin-phone-numbers/src/main/kotlin/com/katas/PhoneList.kt
package com.katas

class PhoneList(private val phoneNumbers: List<String> = emptyList()) {
    fun isConsistent(): Boolean {
        return phoneNumbers.none(::isPrefixOfAnother)
    }

    private fun isPrefixOfAnother(phone: String): Boolean {
        return phoneNumbers
                .filter { it != phone }
                .any { it.contains(phone) }
    }
}
